src/Orleans.Runtime/Messaging/Gateway.cs

```diff
@@ -187,6 +187,11 @@ internal SiloAddress TryToReroute(Message msg)
 
             if (msg.Direction != Message.Directions.Response)
             {
+                if (ClientGrainId.TryParse(msg.SendingGrain, out var clientId) && clients.TryGetValue(clientId, out var clientState))
+                {
+                    clientState.OnClientResponse(msg);
```

Comment:
Isn't it possible that the client is still connected to other silos in the cluster? Or is it an edge case?